polyfills/console/polyfill.js

```diff
@@ -0,0 +1,90 @@
+(function(global) {
+
+	// Grab a referecnce to the console or assign one if it does not exist.
+	var console = global.console || (global.console = {});
+
+	function noop () {}
+
+	// This is based off of angular's internal log factory.
+	function factory(type) {
+
+		var fn = console[type] || console.log || noop;
+		var hasApply = false;
+
+		// Reading `fn.apply` throws in IE11  in IE8 mode. This is because
+		// `typeof console.log === "object"` in IE8
+		try {
+			hasApply = !!fn.apply;
+		} catch (e) { /* do nothing */ }
+
+		// Check if we can safely use `fn.apply`.
+		if (hasApply) {
+			return function () {
+				fn.apply(console, arguments);
```

Comment:
Can't we use `Function.prototype.apply` here rather than checking for `apply` on `fn`?